add package py-lmodule version 0.1.0

shahzebsiddiqui opened the pull request:
I have installed py-lmodule in my spack environment 

```
siddiq90@cori05:/global/u1/s/siddiq90/spack/var/spack/environments/lmodule> spack find
==> In environment lmodule
==> Root specs
py-lmodule 

==> 17 installed packages
-- cray-cnl7-haswell / gcc@9.3.0 --------------------------------
bzip2@1.0.8  gdbm@1.18.1   libbsd@0.10.0  libuuid@1.0.3  openssl@1.1.1g  py-lmodule@0.1.0      python@3.8.5  sqlite@3.31.1  zlib@1.2.11
expat@2.2.9  gettext@0.21  libffi@3.3     ncurses@6.2    pkgconf@1.7.3   py-setuptools@50.1.0  readline@8.0  xz@5.2.5
```

Here is a simple test using the API to ensure package was installed properly
```
siddiq90@cori05:/global/u1/s/siddiq90/spack/var/spack/environments/lmodule> spack load py-lmodule
siddiq90@cori05:/global/u1/s/siddiq90/spack/var/spack/environments/lmodule> which python
/global/u1/s/siddiq90/spack/opt/spack/cray-cnl7-haswell/gcc-9.3.0/python-3.8.5-4742qkh2vmi6ksgwtnc6jyyuyvs5do2l/bin/python
siddiq90@cori05:/global/u1/s/siddiq90/spack/var/spack/environments/lmodule> python
Python 3.8.5 (default, Sep 21 2020, 14:08:29) 
[GCC 9.3.0 20200312 (Cray Inc.)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from lmod.module import Module
>>> cmd = Module("gcc",debug=True)
>>> cmd.test_modules()
[DEBUG] Executing module command: module purge && module load gcc  
[DEBUG] Return Code: 0
0
```

shahzebsiddiqui committed:
add package py-lmodule version 0.1.0

becker33 reviewed (changes requested):
Remove the boilerplate comment

shahzebsiddiqui committed:
add lmod as runtime dependency. Remove boiler plate and update
docstring.
Lmodule is tested with lmod >= 7.x. Lmod 6 has different json
structure in spider which is not supported by lmodule

shahzebsiddiqui commented:
Okay i have installed py-lmodule. 

I realized i missed lmod, so i added this as runtime dependency. I made this assumption given that spack can be used on machine without lmod. I think facility who want to install py-lmodule will use system lmod if its installed.

shahzebsiddiqui committed:
fix flake8 issues

shahzebsiddiqui commented:
@becker33 let me know if you need anything else. This is my first spack package :)
